scripts/assets/options.go.templ

```diff
@@ -0,0 +1,197 @@
+<%
+// EJS template
+// Run `npm run generate:cli-options` to update src/go/rdctl/pkg/options/generated/options.go
+
+
+function collectServerSettingsForJSON(settings, indent, lines) {
+  Object.keys(settings).forEach((propertyName) => {
+    collectServerSettingsForJSONProperty(propertyName, settings[propertyName], indent, lines);
+  });
+}
+
+function collectServerSettingsForJSONProperty(propertyName, settings, indent, lines) {
+  const decln = `${ indent }${ propertyName } `;
+
+  if (typeof (settings.type) === 'object') {
+    lines.push(`${ decln } struct {`);
+    collectServerSettingsForJSON(settings.type, `${ indent }  `, lines);
+    lines.push(`${ indent }} \`json:"${ uncapitalize(propertyName) }"\``);
+  } else {
+    lines.push(`${ decln } *${ settings.type } \`json:"${ uncapitalize(propertyName) },omitempty"\``);
+  }
+}
+
+function collectServerSettingsWithoutJSON(settings, indent, lines) {
+  Object.keys(settings).forEach((propertyName) => {
+    collectServerSettingsPropertyWithoutJSON(propertyName, settings[propertyName], indent, lines);
+  });
+}
+
+function collectServerSettingsPropertyWithoutJSON(propertyName, settings, indent, lines) {
+  const decln = `${ indent }${ propertyName } `;
+
+  if (typeof (settings.type) === 'object') {
+    lines.push(`${ decln } struct {`);
+    collectServerSettingsWithoutJSON(settings.type, `${ indent }  `, lines);
+    lines.push(`${ indent }}`);
+  } else {
+    lines.push(`${ decln } ${ settings.type }`);
+  }
+}
+
+function getUsageForCommand(command) {
+  const usageParts = [command.usageNote ?? ''];
+
+  if (command.enums) {
+    usageParts.push(`(Allowed values: [${ command.enums.join(', ')}].)`);
+  }
+  if (command.aliasFor) {
+    usageParts.push(`(Alias for --${ command.aliasFor }.)`);
+  }
+  return usageParts.join(' ').trim();
+}
+
+function getCommandLineArgValue(lcTypeName, capitalizedName) {
+  switch (lcTypeName) {
+  case 'bool':
+    return `+"="+strconv.FormatBool(SpecifiedSettings.${ capitalizedName })`;
+  case 'int':
+    return `, strconv.Itoa(SpecifiedSettings.${ capitalizedName })`;
+    break;
+  default:
+    return `, SpecifiedSettings.${ capitalizedName }`;
+  }
+}
+
+function uncapitalize(s) {
+  return s[0].toLowerCase() + s.substring(1);
+}
+
+function toGoStringsLiteral(values) {
+  return `[]string{${ values.map(s => JSON.stringify(s) ).join(', ') }}`;
+}
+
+function outputServerSettingsForJSON(serverSettings) {
+  const lines = [];
+
+  collectServerSettingsForJSON(serverSettings, '  ', lines);
+
+  return lines.join('\n');
+}
+
+function outputServerSettingsWithoutJSON(serverSettings) {
+  const lines = [];
+
+  collectServerSettingsWithoutJSON(serverSettings, '  ', lines);
+
+  return lines.join('\n');
+}
+
+%>
+/*
+Copyright © 2022 SUSE LLC
```

Comment:
The template itself should have a license block. We can just move this block to the top so that it serves as the license block for both the template file and the generated file.

Reply:
That's what's happened now, after the code-gen comment.